apps/cowswap-frontend/src/modules/limitOrders/hooks/useUpdateActiveRate.ts

```diff
@@ -17,23 +17,24 @@ export interface UpdateRateCallback {
 export function useUpdateActiveRate(): UpdateRateCallback {
   const { inputCurrencyAmount, outputCurrencyAmount, orderKind } = useLimitOrdersDerivedState()
   const rateState = useAtomValue(limitRateAtom)
-  const updateLimitOrdersState = useSetAtom(updateLimitOrdersRawStateAtom)
+  const updateLimitOrdersState = useUpdateLimitOrdersRawState()
   const updateCurrencyAmount = useUpdateCurrencyAmount()
   const updateRateState = useSetAtom(updateLimitRateAtom)
 
-  const { isRateFromUrl: currentIsRateFromUrl } = rateState
+  const { isRateFromUrl: currentIsRateFromUrl, isAlternativeOrderRate: currentIsAlternativeOrderRate } = rateState
 
   return useCallback(
     (update: RateUpdateParams) => {
-      const { activeRate, isRateFromUrl } = update
+      const { activeRate, isRateFromUrl, isAlternativeOrderRate } = update
 
       updateRateState(update)
 
       const isSell = isSellOrder(orderKind)
 
       if (activeRate) {
         // Don't update amounts when rate is set from URL. See useSetupLimitOrderAmountsFromUrl()
-        if (currentIsRateFromUrl || isRateFromUrl) {
+        // Don't update amounts when rate is set from AlternativeOrder. See AlternativeLimitOrderUpdater
+        if (currentIsRateFromUrl || isRateFromUrl || currentIsAlternativeOrderRate || isAlternativeOrderRate) {
```

Comment:
Maybe this is related to the issue I posted? 
Shouldn't we update the rate for the alternative too?